### ISSUE TITLE
Our app 949539342164783 is getting 500 errors

### ISSUE THREAD

**Issue by cheeseblock** [wit-ai/wit#2129]
**Do you want to request a *feature*, report a *bug*, or ask a *question* about wit?**
Bug
**What is the current behavior?**
App keeps getting 500 errors
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Every time we send a request it mostly returns a 500 error - as with bug #2105 
**What is the expected behavior?**
No 500s
**If applicable, what is the App ID where you are experiencing this issue? If you do not provide this, we cannot help.**
949539342164783

**Comment by cheeseblock** [comment]
I didn't realise there was API versioning - I've updated to the latest API version and it does seem better although I did get one 500 error - I'll keep checking

**Comment by cheeseblock** [comment]
I tried using 20210630 and I'm still getting the 500 errors

**Comment by cheeseblock** [comment]
Also tried 20210513 and got an error - any ideas?

**Comment by CleanestMink126** [comment]
Hi @cheeseblock, just tried some requests and it seemed to be working well. Is the error still occurring? If so, is inference working for you on the website via the "Utterance" input?

**Comment by cheeseblock** [comment]
Hi @CleanestMink126 - thanks for looking into it. Yes I'm still getting the issue. You're testing it by typing phrases into the Utterance box? We're using the API with audio clips - not sure if that matters.
Is there any way of getting detail on the 500 error? Any log I can check? Or any advice you have? Thanks

**Comment by cheeseblock** [comment]
Hi - looking into it further - I tried testing against a previous version/tag of our app - it appears that it doesn't get 500 issues - so it looks as though it's something specific to the app. Is there any way to get any debug info on what might be causing the 500 error? Or any advice on how to dig into the problem? Could our app have got too large and it's a timeout? Thanks

**Comment by chessai** [comment]
> Hi - looking into it further - I tried testing against a previous version/tag of our app - it appears that it doesn't get 500 issues - so it looks as though it's something specific to the app. Is there any way to get any debug info on what might be causing the 500 error? Or any advice on how to dig into the problem? Could our app have got too large and it's a timeout? Thanks

Have you added any builtin entities in the time since the old version?

**Comment by cheeseblock** [comment]
@chessai - thanks for your reply. No I don't believe so.

We are making a bit of progress - we tried exporting and reimporting and there were a few errors. A number of traits were deleted but still referenced and I assume that was causing the 500 errors. Still cleaning things up and need to test - but seems positive.

**Comment by cheeseblock** [comment]
hi @chessai - exporting, importing and fixing any errors that pop up has helped a bit - the 500 errors are now about 1 in 7. In the exported data the version is set to 20200513 - so I tried setting this version in our calls to the server and that has improved things even more - but we still get 500 errors about 1 in 25.

Is there any way of getting extra information about the 500 errors? are you able to see the server logs and check what's going on?

**Comment by cheeseblock** [comment]
@chessai - BTW in the 500 error the downloadHandler has the following text:
{\n  \"code\": \"wit\",\n  \"error\": \"Something went wrong. We've been notified.\"\n}

**Comment by cheeseblock** [comment]
Hi - just a quick bump on this, I wondered if anyone could help. We're due to submit and currently the language processing side doesn't work - so we're a bit stuck. Just want to get some information on the 500 errors we're getting from the server. Thanks

**Comment by chessai** [comment]
> Hi @CleanestMink126 - thanks for looking into it. Yes I'm still getting the issue. You're testing it by typing phrases into the Utterance box? We're using the API with audio clips - not sure if that matters.
> Is there any way of getting detail on the 500 error? Any log I can check? Or any advice you have? Thanks

Are you getting any errors via non-audio? I can't reproduce any errors without audio, haven't tried with audio yet.

**Comment by cheeseblock** [comment]
@chessai - only through audio. Do you have any access to the server logs that might give more info on what's causing the error? or any way that I can access them?

**Comment by chessai** [comment]
> @chessai - only through audio. Do you have any access to the server logs that might give more info on what's causing the error? or any way that I can access them?

No, nothing useful. I'll have to try to reproduce via audio and see what I more debugging info I can produce.

**Comment by cheeseblock** [comment]
Thanks @chessai - appreciate you taking a look.

**Comment by cheeseblock** [comment]
@chessai - any luck reproducing with audio?

**Comment by chessai** [comment]
@cheeseblock no luck. what client are you using? curl, python wit, etc? can you paste the exact code you're calling (with the token obfuscated of course)

**Comment by cheeseblock** [comment]
Thanks for your help @chessai -  We're using Unity web requests. It was working previously - but since adding a number of changes in the wit.ai app it's stopped working - if I run against the old version it still works - this code hasn't changed:

```
private IEnumerator HandleWitAIRequest(WitResponseDelegate callback)
{
	if (!TempVoiceRecorder.HaveValidRecording())
	{
		Debug.LogError("There was no temporary audio to send to Wit.AI. Aborting request.");
		callback?.Invoke(null);
		yield break;
	}

	string requestURL = WIT_AI_VERSION == "" ? WIT_AI_REQUEST_URL : string.Format("{0}?v={1}", WIT_AI_REQUEST_URL, WIT_AI_VERSION); 
	DownloadHandlerBuffer downloadHandler = new DownloadHandlerBuffer();
	UploadHandlerFile uploadHandler = new UploadHandlerFile(TempVoiceRecorder.GetTempAudioPath());
	UnityWebRequest webRequest = new UnityWebRequest(requestURL, "POST", downloadHandler, uploadHandler);
	webRequest.SetRequestHeader("Content-Type", "audio/wav");
	webRequest.SetRequestHeader("Authorization", string.Format("Bearer {0}", WIT_AI_ACCESS_TOKEN));

	// Wait until request is done (we don't need to do anything while we wait, so no loop required)
	yield return webRequest.SendWebRequest();

	if (webRequest.isNetworkError || webRequest.isHttpError)
	{
		Debug.LogErrorFormat("WebRequest Error: {0}", webRequest.error);
		Debug.LogErrorFormat("Response Text: {0}", downloadHandler.text);
		callback.Invoke(null);
	}
	else
	{
...
// log error
...
	}

	TempVoiceRecorder.DeleteRecording();
}
```

**Comment by chainao** [comment]
@chessai, I am facing this same issue (500 Internal Server Error). My app id is 359014025633978 and I am using pywit. The code I am trying to run is the same one that is in the [pywit github](https://github.com/wit-ai/pywit).

```
from wit import Wit
client = Wit(my_token)

resp = None
with open('file.wav', 'rb') as f:
  resp = client.speech(f, {'Content-Type': 'audio/wav'})
print('Yay, got Wit.ai response: ' + str(resp))
```

Could you, please, help me with that?

**Comment by chessai** [comment]
Hi @cheeseblock, are you still encountering issues? We've deployed a fix.

**Comment by cheeseblock** [comment]
@chessai - no luck unfortunately - still the same. Did you manage to reproduce the problem? Thanks

**Comment by cheeseblock** [comment]
hi @chessai - any update on this? Thanks

**Comment by cheeseblock** [comment]
hey @patapizza  @chessai  - we've had this problem now for almost 2 months - we've had to release our app without voice recognition working properly. 

I'm not sure if the devs on wit.ai are volunteers? or whether wit.ai isn't really supposed to be used for commercial purposes? We'd really like an update on this problem - or help fixing it ourselves. Is there a known problem that's currently being worked on? or have you been unable to replicate the problem we are experiencing? Is there anything we can do to help narrow it down. We feel like our hands are tied as it's an internal server error and we have no way of moving forward and not much information being given. I do appreciate any help so far - but it's difficult just to wait patiently with little information.

**Comment by chessai** [comment]
> hey @patapizza @chessai - we've had this problem now for almost 2 months - we've had to release our app without voice recognition working properly.
> 
> I'm not sure if the devs on wit.ai are volunteers? or whether wit.ai isn't really supposed to be used for commercial purposes? We'd really like an update on this problem - or help fixing it ourselves. Is there a known problem that's currently being worked on? or have you been unable to replicate the problem we are experiencing? Is there anything we can do to help narrow it down. We feel like our hands are tied as it's an internal server error and we have no way of moving forward and not much information being given. I do appreciate any help so far - but it's difficult just to wait patiently with little information.

I'm very sorry for your experience. We aren't volunteers - there's a lot that gets worked on behind the scenes, and sometimes catching up with support is difficult given our workload. As for your specific problem, I haven't been able to reproduce with any test apps, so it's been hard to debug. I'm out of the office right now, but on Monday I'm going to attempt a reproduction using your app directly (you shouldn't notice any disruption). If successful, I will then debug. We appreciate your patience and we'll be sure to update you as soon as we have any new information.

**Comment by cheeseblock** [comment]
Thanks very much @chessai - really appreciated.

**Comment by chessai** [comment]
I can reproduce using your app. Debugging now.

**Comment by chessai** [comment]
I have tracked down the failure to an internal service which is returning the 500s, which wit is re-throwing. I am working with the team of that service to figure out what the issue is.

**Comment by cheeseblock** [comment]
Thanks for the update @chessai

**Comment by chessai** [comment]
No updates from the other team yet. Still working on this issue.

**Comment by cheeseblock** [comment]
Hey @chessai - anything back from the other team?

**Comment by chessai** [comment]
> Hey @chessai - anything back from the other team?

Nothing substantial yet. We've found that one of our internal language models is erroring some of the time, which is the cause of your issue. Still working out why and then on a fix.

**Comment by cheeseblock** [comment]
@chessai  - any update? thanks

**Comment by chessai** [comment]
> @chessai - any update? thanks

We have identified a potential issue and are looping more people in to diagnose and work on it. 

Thank you for your patience.

**Comment by chessai** [comment]
We have identified the exact issue and are debugging it.

**Comment by cheeseblock** [comment]
How is the debugging going? Any news?

**Comment by chessai** [comment]
> How is the debugging going? Any news?

Unfortunately the bug is one in PyTorch and needs a lot of legwork to get fixed. We are investigating our options for moving forward.

**Comment by cheeseblock** [comment]
Is there anything we can do on our side to help facilitate a fix?

**Comment by ricvail** [comment]
Hello, I have taken over the development of app 949539342164783 from @cheeseblock . 
May I have an update on the status of this issue?
Thanks

**Comment by chessai** [comment]
> Is there anything we can do on our side to help facilitate a fix?

There is nothing that can be done client-side to facilitate a fix, as far as I am aware.

> Hello, I have taken over the development of app 949539342164783 from @cheeseblock . May I have an update on the status of this issue? Thanks

There are no updates, unfortunately. I have reached out to the team working on the issue and am waiting on a response.

**Comment by ricvail** [comment]
Hello,
I opened a new issue, since I'm not sure if we're still experiencing the same problem or it's a different one: https://github.com/wit-ai/wit/issues/2247
Either way the new issue contains some more information.

**Comment by ramh** [comment]
@ricvail I am just cleaning up older issues. Do you think it is ok to close this issue in favor of #2247  ?

**Comment by ricvail** [comment]
@ramh Yes, go ahead

**Comment by ramh** [comment]
Thanks. Closing in favor of #2247 .

**Comment by lolatafoureau** [comment]
Hi @chessai 
My app 625337711941868 is getting 500 ERROR. Could you help me to fix this please ?
Thanks
